### PR TITLE
Add 'jwt-cpp' package

### PR DESCRIPTION
**Pull request by Thalhammer**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

### CODE REVIEW

**Review by rbsheth — APPROVED** [review]

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -84,6 +84,7 @@ hunter_default_version(Imath VERSION 3.1.1)
 hunter_default_version(Immer VERSION 0.6.2-cf44615)
 hunter_default_version(Jpeg VERSION 9c-p0)
 hunter_default_version(JsonSpirit VERSION 0.0.4-hunter)
+hunter_default_version(jwt-cpp VERSION 0.5.2)
```
Comment (rbsheth) [inline cmake/configs/default.cmake:87]: Looks like the alphabetical order checker isn't happy. Can you move this to the lower-case list later in this file?

**Comment by rbsheth** [comment]
@Thalhammer looks like the install step is failing on Unix-based systems - are you able to debug?

**Comment by Thalhammer** [comment]
> @Thalhammer looks like the install step is failing on Unix-based systems - are you able to debug?

I am a bit surprised by this tbh since we have that code working in multiple peoples production environments without issue.

I'll take a look at what's happening later.

**Comment by prince-chrismc** [comment]
Looks like an issue with the hunter integration https://github.com/cpp-pm/hunter/pull/467/checks?check_run_id=3546137414#step:7:3914

Might be related to https://github.com/Thalhammer/jwt-cpp/blob/aaf3c714e343dc381a5dfca921075b27f89d6a2c/CMakeLists.txt#L75-L76 but I am not sure 😖

**Comment by NeroBurner** [comment]
https://github.com/Thalhammer/jwt-cpp/blob/aaf3c714e343dc381a5dfca921075b27f89d6a2c/CMakeLists.txt#L91

the `JWT_CMAKE_FILES_INSTALL_DIR` variable should NOT be an absolute path. Make it just

```
set(JWT_CMAKE_FILES_INSTALL_DIR cmake)
```

otherwise the generated `jwt-config.cmake` is not relocatable, which hunter needs to create cached packages (or use a cache server)

**Comment by Thalhammer** [comment]
Looks like @NeroBurner is right. If I remove the install prefix it still uses the install prefix on installation but the generated jwt-cpp-targets.cmake only has relative paths.

**Comment by rbsheth** [comment]
Looks like things are still failing... any updates?

**Review by rbsheth — APPROVED** [review]

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.315

**Pull request merged**